src/meson.build

```diff
@@ -14,19 +16,26 @@ sources += [
     'nvme/log.c',
     'nvme/tree.c',
     'nvme/util.c',
+    join_paths(source_dir, '../ccan/ccan/list/list.c'),
+    join_paths(source_dir, '../ccan/ccan/str/debug.c'),
+    join_paths(source_dir, '../ccan/ccan/str/str.c'),
```

Comment:
You don't really need to join these paths to the source dir.

What is the difference between `$PWD/../ccan/ccan/list/list.c` and `../ccan/ccan/list/list.c`? Meson processes them as files either way, and resolves them relative to current_source_dir.

There is a better reason to use it for the version-script. Meson doesn't know that that link argument is a file, so it cannot relativize it.